spec.html

```diff
@@ -34624,7 +34823,34 @@ <h1>Static Semantics: Early Errors</h1>
         <emu-grammar>UnicodePropertyValueExpression :: LoneUnicodePropertyNameOrValue</emu-grammar>
         <ul>
           <li>
-            It is a Syntax Error if the List of Unicode code points that is SourceText of |LoneUnicodePropertyNameOrValue| is not identical to a List of Unicode code points that is a Unicode general category or general category alias listed in the &ldquo;Property value and aliases&rdquo; column of <emu-xref href="#table-unicode-general-category-values"></emu-xref>, nor a binary property or binary property alias listed in the &ldquo;Property name and aliases&rdquo; column of <emu-xref href="#table-binary-unicode-properties"></emu-xref>.
+            It is a Syntax Error if the List of Unicode code points that is SourceText of |LoneUnicodePropertyNameOrValue| is not identical to a List of Unicode code points that is a Unicode general category or general category alias listed in the &ldquo;Property value and aliases&rdquo; column of <emu-xref href="#table-unicode-general-category-values"></emu-xref>, nor a binary property or binary property alias listed in the &ldquo;Property name and aliases&rdquo; column of <emu-xref href="#table-binary-unicode-properties"></emu-xref>, nor a binary property of strings listed in the &ldquo;Property name&rdquo; column of <emu-xref href="#table-binary-unicode-properties-of-strings"></emu-xref>.
+          </li>
+          <li>
+            It is a Syntax Error if the enclosing |Pattern| does not have a [UnicodeSetsMode] parameter and the List of Unicode code points that is SourceText of |LoneUnicodePropertyNameOrValue| is identical to a List of Unicode code points that is a binary property of strings listed in the &ldquo;Property name&rdquo; column of <emu-xref href="#table-binary-unicode-properties-of-strings"></emu-xref>.
+          </li>
+        </ul>
+        <emu-grammar>CharacterClassEscape :: `P{` UnicodePropertyValueExpression `}`</emu-grammar>
+        <ul>
+          <li>
+            It is a Syntax Error if MaybeStrings of the |UnicodePropertyValueExpression| is *true*.
+          </li>
+        </ul>
+        <emu-grammar>CharacterClass[UnicodeMode, UnicodeSetsMode] :: `[` `^` ClassRanges[?UnicodeMode, ?UnicodeSetsMode] `]`</emu-grammar>
```

Comment:
```suggestion
        <emu-grammar>CharacterClass :: `[` `^` ClassRanges `]`</emu-grammar>
```
We normally don't put grammatical parameters in a non-defining production unless the accompanying prose refers to it/them.

(Similarly throughout.)

Reply:
Done here and below